src/Compilers/CSharp/Test/Semantic/Semantics/StructConstructorTests.cs

```diff
@@ -0,0 +1,1329 @@
+@"#nullable enable
+struct S0
+{
+    object F0 = Utils.GetValue();
+}
+struct S1
+{
+    object F1 = Utils.GetValue();
+    public S1() { }
+}
+struct S2
+{
+    object F2 = Utils.GetValue();
+    public S2() : this(null) { }
+    S2(object obj) { }
+}
+static class Utils
+{
+    internal static object? GetValue() => null;
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(
+                // (4,17): warning CS8601: Possible null reference assignment.
+                //     object F0 = Utils.GetValue();
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "Utils.GetValue()").WithLocation(4, 17),
+                // (8,17): warning CS8601: Possible null reference assignment.
+                //     object F1 = Utils.GetValue();
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "Utils.GetValue()").WithLocation(8, 17),
+                // (13,17): warning CS8601: Possible null reference assignment.
+                //     object F2 = Utils.GetValue();
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "Utils.GetValue()").WithLocation(13, 17),
+                // (14,24): warning CS8625: Cannot convert null literal to non-nullable reference type.
+                //     public S2() : this(null) { }
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(14, 24));
+        }
+
+        [Fact]
+        public void DefiniteAssignment()
+        {
+            var source =
+@"#pragma warning disable 169
+unsafe struct S0
+{
+    fixed int Y[1];
+}
+unsafe struct S1
+{
+    fixed int Y[1];
+    public S1() { }
+}
+unsafe struct S2
+{
+    int X;
+    fixed int Y[1];
+}
+unsafe struct S3
+{
+    int X;
+    fixed int Y[1];
+    public S3() { }
+}
+unsafe struct S4
+{
+    int X = 4;
+    fixed int Y[1];
+}
+unsafe struct S5
+{
+    int X;
+    fixed int Y[1];
+    public S5() { X = 5; }
+}";
+
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview, options: TestOptions.UnsafeReleaseDll);
+            comp.VerifyDiagnostics(
+                // (20,12): error CS0171: Field 'S3.X' must be fully assigned before control is returned to the caller
+                //     public S3() { }
+                Diagnostic(ErrorCode.ERR_UnassignedThis, "S3").WithArguments("S3.X").WithLocation(20, 12),
+                // (24,9): warning CS0414: The field 'S4.X' is assigned but its value is never used
+                //     int X = 4;
+                Diagnostic(ErrorCode.WRN_UnreferencedFieldAssg, "X").WithArguments("S4.X").WithLocation(24, 9),
+                // (29,9): warning CS0414: The field 'S5.X' is assigned but its value is never used
+                //     int X;
+                Diagnostic(ErrorCode.WRN_UnreferencedFieldAssg, "X").WithArguments("S5.X").WithLocation(29, 9));
+        }
+
+        [Fact]
+        public void ParameterDefaultValues_01()
```

Comment:
If I recall correctly, there's some subtle differences between what's allowed for default parameter values and constants. Consider adding some tests for constants.